examples/custom-inspector/src/main.rs

```diff
@@ -0,0 +1,138 @@
+//! Example of how to use a custom inspector to trace new pending transactions
+//!
+//! Run with
+//!
+//! ```not_rust
+//! cargo run --release -p custom-inspector --node --http --ws --recipients 0x....,0x....
+//! ```
+//!
+//! If no recipients are specified, all transactions will be inspected.
+
+use clap::Parser;
+use futures_util::StreamExt;
+use reth::{
+    cli::{
+        components::{RethNodeComponents, RethRpcComponents, RethRpcServerHandles},
+        config::RethRpcConfig,
+        ext::{RethCliExt, RethNodeCommandConfig},
+        Cli,
+    },
+    primitives::{Address, BlockId, IntoRecoveredTransaction},
+    revm::{
+        inspector_handle_register, interpreter::Interpreter, revm, Database, EvmContext, Inspector,
+    },
+    rpc::{
+        compat::transaction::transaction_to_call_request,
+        eth::{revm_utils::EvmOverrides, EthTransactions},
+    },
+    tasks::TaskSpawner,
+    transaction_pool::TransactionPool,
+};
+use reth_revm::interpreter::OpCode;
+use std::collections::HashSet;
+
+fn main() {
+    Cli::<MyRethCliExt>::parse().run().unwrap();
+}
+
+/// The type that tells the reth CLI what extensions to use
+struct MyRethCliExt;
+
+impl RethCliExt for MyRethCliExt {
+    /// This tells the reth CLI to trace addresses via `RethCliTxpoolExt`
+    type Node = RethCliTxpoolExt;
+}
+
+/// Our custom cli args extension that adds one flag to reth default CLI.
+#[derive(Debug, Clone, Default, clap::Args)]
+struct RethCliTxpoolExt {
+    /// recipients addresses that we want to trace
+    #[arg(long, value_delimiter = ',')]
+    pub recipients: Vec<Address>,
+}
+
+#[derive(Default, Debug, Clone)]
+struct DummyInspector {
+    ret_val: Vec<String>,
+}
```

Comment:
we should add a smol doc here what this inspector does, listing opcodes at the program counter

Reply:
Sure, will do in tomorrow.

Reply:
Done.